habitat-baselines/habitat_baselines/rl/ppo/trainer_agent.py

```diff
@@ -0,0 +1,333 @@
+            actor_critic.load_state_dict(
+                {  # type: ignore
+                    k[len("actor_critic.") :]: v
+                    for k, v in pretrained_state["state_dict"].items()
+                }
+            )
+        elif self._config.habitat_baselines.rl.ddppo.pretrained_encoder:
+            prefix = "actor_critic.net.visual_encoder."
+            actor_critic.net.visual_encoder.load_state_dict(
+                {
+                    k[len(prefix) :]: v
+                    for k, v in pretrained_state["state_dict"].items()
+                    if k.startswith(prefix)
+                }
+            )
+        if self._is_static_encoder:
+            for param in actor_critic.net.visual_encoder.parameters():
+                param.requires_grad_(False)
+
+        return actor_critic
+
+    @property
+    def rollouts(self) -> RolloutStorage:
+        return self._rollouts
+
+    @property
+    def actor_critic(self) -> NetPolicy:
+        return self._actor_critic
+
+    @property
+    def updater(self) -> PPO:
+        return self._updater
+
+    def get_resume_state(self) -> Dict[str, Any]:
+        return dict(
+            state_dict=self._actor_critic.state_dict(),
+            optim_state=self._updater.optimizer.state_dict(),
+            lr_sched_state=self._lr_scheduler.state_dict(),
+        )
+
+    def get_save_state(self) -> Dict:
+        return dict(state_dict=self.actor_critic.state_dict())
+
+    def eval(self):
+        self.actor_critic.eval()
+
+    def train(self):
+        self.actor_critic.train()
+        self.updater.train()
+
+    def load_ckpt_state_dict(self, ckpt: Dict) -> None:
+        """
+        Loads a state dict for evaluation. The difference from
+        `load_state_dict` is that this will not load the policy state if the
+        policy does not request it.
+        """
+        self.actor_critic.load_state_dict(ckpt["state_dict"])
+
+    def load_state_dict(self, state: Dict) -> None:
+        self._actor_critic.load_state_dict(state["state_dict"])
+        if self._updater is not None:
+            if "optim_state" in state:
+                self._actor_critic.load_state_dict(state["optim_state"])
+            if "lr_sched_state" in state:
+                self._actor_critic.load_state_dict(state["lr_sched_state"])
+
+    @property
+    def hidden_state_shape(self):
+        """
+        The shape of the tensor to track the hidden state, such as the RNN hidden state.
+        """
+
+        return (
+            self.actor_critic.num_recurrent_layers,
+            self._ppo_cfg.hidden_size,
+        )
+
+    def after_update(self):
+        """
+        Called after the updater has called `update` and the rollout `after_update` is called.
+        """
+
+        if self._ppo_cfg.use_linear_lr_decay:
+            self._lr_scheduler.step()  # type: ignore
+
+    def pre_rollout(self):
+        """
+        Called before a rollout is collected.
+        """
```

Comment:
Same comment as above, this is manually called but sounds like it is a callback.
I would call this `update_clip_parameter` or even better call this automatically.